archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterTest.java

```diff
@@ -1857,6 +1871,115 @@ public void is_resilient_against_broken_ClassFileSources() throws MalformedURLEx
         assertThat(classes).isEmpty();
     }
 
+    @Test
+    public void synthetic_bridge_methods() throws Exception {
+        final String prefix = "com.tngtech.archunit.core.importer.testexamples.syntheticbridge";
+
+        final ImportedClasses javaClasses = classesIn("testexamples/syntheticbridge");
+
+        final JavaClass javaClassBase = javaClasses.get(SyntheticBridgeMethodsClass.class.getSuperclass());
+        assertThat(javaClassBase.getMethods())
+                .hasSize(5)
+                .extracting(extractFullName())
+                .contains(prefix + ".BasePackageProtected.privateMethod()")
+                .contains(prefix + ".BasePackageProtected.protectedMethod()")
+                .contains(prefix + ".BasePackageProtected.hashCode()")
+                .contains(prefix + ".BasePackageProtected.publicMethod()")
+                .contains(prefix + ".BasePackageProtected.packageProtectedMethod()");
+
+        final JavaClass javaClass = javaClasses.get(SyntheticBridgeMethodsClass.class);
+        assertThat(javaClass.getMethods())
+                .isEmpty();
+    }
+
+    @Test
+    @Ignore
+    public void synthetic_constructor() throws Exception {
+        final String prefix = "com.tngtech.archunit.core.importer.testexamples.syntheticbridge";
+
+        final ImportedClasses javaClasses = classesIn("testexamples/syntheticbridge");
+        final JavaClass javaClass = javaClasses.get(SyntheticConstructor.class);
+        assertThat(javaClass.getConstructors())
+                .extracting(extractFullName())
+                .contains(prefix + ".SyntheticConstructor.<init>()")
+                .doesNotContain(prefix + ".SyntheticConstructor.<init>(" + prefix + ".SyntheticConstructor$1)")
+                .hasSize(1);
+    }
+
+    @Test
+    public void synthetic_field() throws Exception {
+        final String prefix = "com.tngtech.archunit.core.importer.testexamples.syntheticbridge";
+
+        final ImportedClasses javaClasses = classesIn("testexamples/syntheticbridge");
+        final JavaClass javaClass = javaClasses.get(FooEnum.class);
+        assertThat(javaClass.getFields())
+                .extracting(extractFullName())
+                .contains(prefix + ".FooEnum.X")
+                .contains(prefix + ".FooEnum.Y")
+                .contains(prefix + ".FooEnum.Z")
+                .doesNotContain(prefix + ".FooEnum.$VALUES")
+                .hasSize(3);
+    }
+
+    @Test
+    public void synthetic_class() throws Exception {
+        thrown.expect(NullPointerException.class);
+        thrown.expectMessage(containsString("No object with name"));
+
+        final ImportedClasses javaClasses = classesIn("testexamples/syntheticbridge");
+        javaClasses.get("com.tngtech.archunit.core.importer.testexamples.syntheticbridge.SyntheticConstructor$1");
+    }
+
+    @Test
+    public void synthetic_class_switch_map() throws Exception {
```

Comment:
Found testcase when synthetic switchmap class is created. It is ok with current changes, class is loaded into com.tngtech.archunit.core.importer.ImportedClasses#additionalClasses and correctly skipped.